src/EA.Weee.RequestHandlers.Tests.Unit/AatfReturn/AddReturnSchemeHandlerTests.cs

```diff
@@ -0,0 +1,109 @@
+        }
+
+        [Fact]
+        public async Task HandleAsync_NoExternalAccess_ThrowsSecurityException()
+        {
+            var authorization = new AuthorizationBuilder().DenyExternalAreaAccess().Build();
+
+            handler = new AddReturnSchemeHandler(authorization, A.Dummy<IReturnSchemeDataAccess>(), A.Dummy<IReturnDataAccess>(), A.Dummy<ISchemeDataAccess>());
+
+            Func<Task> action = async () => await handler.HandleAsync(A.Dummy<AddReturnScheme>());
+
+            await action.Should().ThrowAsync<SecurityException>();
+        }
+
+        [Fact]
+        public async Task HandleAsync_GivenAddReturnSchemeRequest_SchemeShouldBeRetrieved()
+        {
+            var schemeId = Guid.NewGuid();
+
+            var request = new AddReturnScheme { SchemeId = schemeId };
+
+            await handler.HandleAsync(request);
+
+            A.CallTo(() => schemeDataAccess.GetSchemeOrDefault(schemeId)).MustHaveHappened(Repeated.Exactly.Once);
+        }
+
+        [Fact]
+        public async Task HandleAsync_GivenAddReturnSchemeRequest_ReturnShouldBeRetrieved()
+        {
+            var returnId = Guid.NewGuid();
+
+            var request = new AddReturnScheme { ReturnId = returnId };
+
+            await handler.HandleAsync(request);
+
+            A.CallTo(() => returnDataAccess.GetById(returnId)).MustHaveHappened(Repeated.Exactly.Once);
+        }
+
+        [Fact]
+        public async Task HandleAsync_GivenAddReturnSchemeRequest_DataAccessSubmitsIsCalled()
+        {
+            var request = new AddReturnScheme { ReturnId = Guid.NewGuid(), SchemeId = Guid.NewGuid() };
+
+            var @return = A.Dummy<Return>();
+            var scheme = A.Fake<Domain.Scheme.Scheme>();
+
+            A.CallTo(() => schemeDataAccess.GetSchemeOrDefault(request.SchemeId)).Returns(scheme);
+
+            A.CallTo(() => returnDataAccess.GetById(request.ReturnId)).Returns(@return);
+
+            await handler.HandleAsync(request);
+            
+            A.CallTo(() => returnSchemeDataAccess.Submit(A<ReturnScheme>.That.Matches(c => c.ReturnId == @return.Id && c.SchemeId == scheme.Id))).MustHaveHappened(Repeated.Exactly.Once);
+        }
+
+        [Fact]
+        public async Task HandleAsync_GivenAddReturnSchemeRequest_ReturnSchemeIdShouldBeReturned()
+        {
+            var request = new AddReturnScheme { ReturnId = Guid.NewGuid(), SchemeId = Guid.NewGuid() };
+
+            var @return = A.Dummy<Return>();
+            var scheme = A.Fake<Domain.Scheme.Scheme>();
+
+            A.CallTo(() => schemeDataAccess.GetSchemeOrDefault(request.SchemeId)).Returns(scheme);
+
+            A.CallTo(() => returnDataAccess.GetById(request.ReturnId)).Returns(@return);
+
+            var result = await handler.HandleAsync(request);
+
+            A.CallTo(() => returnSchemeDataAccess.Submit(A<ReturnScheme>.That.Matches(c => c.ReturnId == @return.Id && c.SchemeId == scheme.Id))).Returns(returnSchemeId);
```

Comment:
Rather than A.CallTo(() => returnSchemeDataAccess.Submit(A<ReturnScheme>.That.Matches(c => c.ReturnId == @return.Id && c.SchemeId == scheme.Id))).Returns(returnSchemeId);

Should be able to do result.Should().be(request.ReturnId)